src/plugins/spherical/spherical.js

```diff
@@ -0,0 +1,147 @@
+ *
+ * This plugin provides support for positioning steps with polar or shperical coordinates.
+ * This is often more convenient when arranging slides along a circle or similar. It also
+ * supports relative positioning.
+ *
+ * Example:
+ *
+ *  <!-- Position step 1000 px distance to origin at an angle of 50 degree. -->
+ *  <div class="step" data-spherical-dist="1000" data-spherical-rho="50">
+ *
+ *  <!-- Position step 500 px distance to origin at a primary angle of 270 degree and a
+ *       secondary angle of -30 degree (= 330 degree) -->
+ *  <div class="step" data-spherical-dist="500" data-spherical-rho="270" data-spherical-theta="-30">
+ *
+ *
+ * Following html attributes are supported for step elements:
+ *
+ *     data-spherical-distance
+ *     data-spherical-rho
+ *     data-spherical-theta
+ *
+ *     data-spherical-rel-distance
+ *     data-sphercial-rel-rho
+ *     data-spherical-rel-theta
+ *
+ * Non-zero relative values are inherited from the previous step. This makes it easy to
+ * create a presentation where subsequent slides are arranged along a circle.
+ *
+ * The above spherical values are ignored, or set to zero, if the corresponding
+ * absolute values (`data-x` etc...) is set.
+ *
+ * The plugin translates all spherical coordinates to cartesian coordinates, which then could
+ * for example be used by the `rel` plugin. In fact, to get the relative positioning working,
+ * the `rel` plugin is needed, actually. If it is not present, relative values are silently ignored.
+ *
+ * Copyright 2018 Holger Teichert (@complanar)
+ * Released under the MIT license.
+ */
+
+/* global document, window */
+
+( function( document, window ) {
+    "use strict";
+
+    var attributeTracker = [];
+
+    /**
+     * Copied from core impress.js. We currently lack a library mechanism to
+     * to share utility functions like this.
+     */
+    var toNumber = function( numeric, fallback ) {
+        return isNaN( numeric ) ? ( fallback || 0 ) : Number( numeric );
+    };
+
+    var sphericalToCartesian = function( r, phi, theta ) {
+        var x = r * Math.sin( theta ) * Math.cos( phi );
+        var y = r * Math.sin( theta ) * Math.sin( phi );
+        var z = r * Math.cos( theta );
+        return { x: x, y: y, z: z };
+    };
+
+    var computeCartesianPositions = function( el ) {
+        var cartesian, data;
+
+        data = el.dataset;
+
+        if ( data.sphericalDistance && data.sphericalPhi ) {
+            if ( !data.sphericalTheta ) {
+                attributeTracker.push( { "node": el, "attr": "data-spherical-theta" } );
+                el.setAttribute( "data-spherical-theta", 90 );
+            }
+
+            cartesian = sphericalToCartesian(
+                toNumber( data.sphericalDistance, 0 ),
+                toNumber( data.sphericalPhi * Math.PI / 180, 0 ),
+                toNumber( data.sphericalTheta * Math.PI / 180, 90 )
+            );
+
+            if ( data.x === undefined ) {
```

Comment:
I don't think this works. Use of spherical coordinates will generally always set all of x/y/z. Therefore you cannot check and set them one by one, rather it is all or nothing:

* If any data-spherical-* attributes are set, then there must be no data-rel-x/y/z and no data-x/y/z attributes set. 
* If even one of them is set, then all data-spherical-* attributes are ignored and a warning logged.